mkl error && make test failed with icpx && onemkl

#I'm compiling cvode-6.5.0 with icx && icpx for sundials_mkldense based on oneapi-2023.0.0 
- with these options:
````cmake
cmake \
-DCMAKE_INSTALL_PREFIX=/home/ljl/Apps/cvode \
-DCMAKE_BUILD_TYPE=Debug \
-DCMAKE_C_COMPILER=icx \
-DCMAKE_CXX_COMPILER=icpx \
-DENABLE_ONEMKL=ON \
-DENABLE_SYCL=ON \
-DBUILD_TESTING=ON \
-DEXAMPLES_ENABLE_C=OFF \
-DONEMKL_DIR=/opt/intel/oneapi/mkl/2023.0.0 \
..
````
##for <CL/sycl.hpp> header error ,I add "-fsycl" for CMAKE_CXX_FLAGS in ./CVODE-6.5.0/cmake/SundialsSetupCompilers.cmake
````cmake
if(ENABLE_SYCL)#line 27
  set(DOCSTR "The C++ standard to use if C++ is enabled (17, 20)")
  sundials_option(CMAKE_CXX_STANDARD STRING "${DOCSTR}" "17"
                  OPTIONS "17;20")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl")
else()
  set(DOCSTR "The C++ standard to use if C++ is enabled (14, 17, 20)")
  sundials_option(CMAKE_CXX_STANDARD STRING "${DOCSTR}" "14"
                  OPTIONS "14;17;20")
endif()
````
##then make -j failed with error:
````bash
icx: error: invalid argument '-std=gnu99' not allowed with '-fsycl'
````
-  I found the reason: find_package(MKL) seems add flags "-fsycl" for c compiler in line 419 which makes confilct to "-std=gnu99"
##Commite either options "CMAKE_C_STANDARD" in SundialsSetupCompilers.cmake or line-419 in MKLConfig.cmake makes errors while making test :
````bash
The following tests FAILED:
          5 - test_nvector_sycl_7_32_0 (Subprocess aborted)
          6 - test_nvector_sycl_500_128_0 (Subprocess aborted)
          7 - test_nvector_sycl_1000_0_0 (Subprocess aborted)
         21 - test_sunmatrix_onemkldense_100_100_1_0 (Subprocess aborted)
         22 - test_sunmatrix_onemkldense_200_1000_1_0 (Subprocess aborted)
         23 - test_sunmatrix_onemkldense_2000_100_1_0 (Subprocess aborted)
         24 - test_sunmatrix_onemkldense_10_10_100_0 (Subprocess aborted)
         25 - test_sunmatrix_onemkldense_200_10_100_0 (Subprocess aborted)
         26 - test_sunmatrix_onemkldense_10_200_100_0 (Subprocess aborted)
         46 - test_sunlinsol_onemkldense_10_1_0 (Subprocess aborted)
         47 - test_sunlinsol_onemkldense_100_1_0 (Subprocess aborted)
         48 - test_sunlinsol_onemkldense_10_1000_0 (Subprocess aborted)
         49 - test_sunlinsol_onemkldense_100_100_0 (Subprocess aborted)
Errors while running CTest
Output from these tests are in: /home/ljl/Downloads/cvode-6.5.0/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make: *** [Makefile:74: test] Error 8
```` 
###check well please

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # SUNDIALS Changelog
 
+## Changes to SUNDIALS in release 6.6.0
+
+Fixed compilation errors in some SYCL examples when using the `icx` compiler.
+
 ## Changes to SUNDIALS in release 6.5.0
 
 Added the functions `ARKStepGetJac`, `ARKStepGetJacTime`,

--- a/doc/arkode/guide/source/Introduction.rst
+++ b/doc/arkode/guide/source/Introduction.rst
@@ -118,6 +118,11 @@ provided with SUNDIALS, or again may utilize a user-supplied module.
 Changes from previous versions
 ==============================
 
+Changes in v5.6.0
+-----------------
+
+Fixed compilation errors in some SYCL examples when using the ``icx`` compiler.
+
 Changes in v5.5.0
 -----------------
 

--- a/doc/cvode/guide/source/Introduction.rst
+++ b/doc/cvode/guide/source/Introduction.rst
@@ -111,6 +111,11 @@ implementations.
 Changes from previous versions
 ==============================
 
+Changes in v6.6.0
+-----------------
+
+Fixed compilation errors in some SYCL examples when using the ``icx`` compiler.
+
 Changes in v6.5.0
 -----------------
 

--- a/doc/cvodes/guide/source/Introduction.rst
+++ b/doc/cvodes/guide/source/Introduction.rst
@@ -111,6 +111,11 @@ Fortran.
 Changes from previous versions
 ==============================
 
+Changes in v6.6.0
+-----------------
+
+Fixed compilation errors in some SYCL examples when using the ``icx`` compiler.
+
 Changes in v6.5.0
 -----------------
 

--- a/doc/ida/guide/source/Introduction.rst
+++ b/doc/ida/guide/source/Introduction.rst
@@ -72,6 +72,11 @@ systems.
 Changes from previous versions
 ==============================
 
+Changes in v6.6.0
+-----------------
+
+Fixed compilation errors in some SYCL examples when using the ``icx`` compiler.
+
 Changes in v6.5.0
 -----------------
 

--- a/doc/idas/guide/source/Introduction.rst
+++ b/doc/idas/guide/source/Introduction.rst
@@ -86,6 +86,11 @@ integrate any final-condition ODE dependent on the solution of the original IVP
 Changes from previous versions
 ==============================
 
+Changes in v5.6.0
+-----------------
+
+Fixed compilation errors in some SYCL examples when using the ``icx`` compiler.
+
 Changes in v5.5.0
 -----------------
 

--- a/doc/kinsol/guide/source/Introduction.rst
+++ b/doc/kinsol/guide/source/Introduction.rst
@@ -88,6 +88,11 @@ applications written in Fortran.
 Changes from previous versions
 ==============================
 
+Changes in v6.6.0
+-----------------
+
+Fixed compilation errors in some SYCL examples when using the ``icx`` compiler.
+
 Changes in v6.5.0
 -----------------
 

--- a/doc/shared/Install.rst
+++ b/doc/shared/Install.rst
@@ -938,10 +938,11 @@ illustration only.
 
    .. note::
 
-      At present the only supported SYCL compiler is the DPC++ (Intel oneAPI)
-      compiler. CMake does not currently support autodetection of SYCL compilers
-      and ``CMAKE_CXX_COMPILER`` must be set to a valid SYCL compiler i.e.,
-      ``dpcpp`` in order to build with SYCL support.
+      CMake does not currently support autodetection of SYCL compilers and
+      ``CMAKE_CXX_COMPILER`` must be set to a valid SYCL compiler. At present
+      the only supported SYCL compilers are the Intel oneAPI compilers i.e.,
+      ``dpcpp`` and ``icpx``. When using ``icpx`` the ``-fsycl`` flag and any
+      ahead of time compilation flags must be added to ``CMAKE_CXX_FLAGS``.
 
 
 .. cmakeoption:: SUNDIALS_LOGGING_LEVEL

--- a/examples/sunlinsol/CMakeLists.txt
+++ b/examples/sunlinsol/CMakeLists.txt
@@ -32,6 +32,10 @@ add_subdirectory(pcg/serial)
 
 # Build the sunlinsol test utilities
 add_library(test_sunlinsol_obj OBJECT test_sunlinsol.c test_sunlinsol.h)
+if(BUILD_SHARED_LIBS)
+  # need PIC when shared libs are used since the example executables will link to the shared lib
+  set_property(TARGET test_sunlinsol_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
+endif()
 target_link_libraries(test_sunlinsol_obj PRIVATE sundials_sunlinsoldense)
 
 if(ENABLE_MPI AND MPI_C_FOUND)

--- a/examples/sunlinsol/onemkldense/CMakeLists.txt
+++ b/examples/sunlinsol/onemkldense/CMakeLists.txt
@@ -40,7 +40,7 @@ foreach(example_tuple ${sunlinsols_onemkldense_examples})
   if (NOT TARGET ${example_target})
 
     # example source files
-    add_executable(${example_target} ${example} ../test_sunlinsol.c ../test_sunlinsol.h)
+    add_executable(${example_target} ${example})
 
     # folder for IDEs
     set_target_properties(${example_target} PROPERTIES FOLDER "Examples")
@@ -48,6 +48,7 @@ foreach(example_tuple ${sunlinsols_onemkldense_examples})
     # libraries to link against
     target_link_libraries(${example_target}
       PRIVATE
+      test_sunlinsol_obj
       sundials_nvecsycl
       sundials_sunlinsolonemkldense
       MKL::MKL_DPCPP

--- a/examples/sunmatrix/CMakeLists.txt
+++ b/examples/sunmatrix/CMakeLists.txt
@@ -27,6 +27,10 @@ add_subdirectory(sparse)
 
 # Build the sunmatrix test utilities
 add_library(test_sunmatrix_obj OBJECT test_sunmatrix.c test_sunmatrix.h)
+if(BUILD_SHARED_LIBS)
+  # need PIC when shared libs are used since the example executables will link to the shared lib
+  set_property(TARGET test_sunmatrix_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
+endif()
 target_link_libraries(test_sunmatrix_obj PRIVATE sundials_sunmatrixdense)
 
 if(BUILD_SUNMATRIX_CUSPARSE)

--- a/examples/sunmatrix/onemkldense/CMakeLists.txt
+++ b/examples/sunmatrix/onemkldense/CMakeLists.txt
@@ -42,7 +42,7 @@ foreach(example_tuple ${sunmatrix_onemkldense_examples})
   if (NOT TARGET ${example_target})
 
     # example source files
-    add_executable(${example_target} ${example} ../test_sunmatrix.c ../test_sunmatrix.h)
+    add_executable(${example_target} ${example})
 
     # folder for IDEs
     set_target_properties(${example_target} PROPERTIES FOLDER "Examples")
@@ -50,6 +50,7 @@ foreach(example_tuple ${sunmatrix_onemkldense_examples})
     # libraries to link against
     target_link_libraries(${example_target}
       PRIVATE
+      test_sunmatrix_obj
       sundials_nvecsycl
       sundials_sunmatrixonemkldense
       MKL::MKL_DPCPP

